testing/src/test/java/io/stargate/it/cql/BoundStatementTest.java

```diff
@@ -0,0 +1,172 @@
+  @Test
+  public void should_allow_nulls_values() {
+    PreparedStatement prepared = session.prepare("INSERT INTO test2 (k, v0) values (?, ?)");
+    session.execute(prepared.bind(KEY, null));
+
+    Row row = session.execute("SELECT v0 FROM test2 WHERE k=?", KEY).one();
+    assertThat(row.isNull("v0")).isTrue();
+  }
+
+  @Test
+  @Disabled("Returns an empty result set instead of failing")
+  public void should_fail_if_missing_values() {
+    PreparedStatement prepared = session.prepare("SELECT v FROM test3 WHERE pk1=? and pk2=?");
+    assertThatThrownBy(() -> session.execute(prepared.bind(1)))
+        .isInstanceOf(InvalidQueryException.class);
+  }
+
+  @Test
+  @Disabled("Doesn't seem to work at the moment")
+  public void should_not_write_tombstone_if_value_is_implicitly_unset() {
+    PreparedStatement prepared = session.prepare("INSERT INTO test2 (k, v0) values (?, ?)");
+    session.execute(prepared.bind(KEY, VALUE));
+
+    BoundStatement boundStatement = prepared.bind(KEY);
+    assertThat(boundStatement.isSet("v0")).isFalse();
+    session.execute(boundStatement);
+
+    Row row = session.execute("SELECT v0 FROM test2 WHERE k=?", KEY).one();
+    assertThat(row.getInt("v0")).isEqualTo(VALUE);
+  }
+
+  @Test
+  @Disabled("Doesn't seem to work at the moment")
+  public void should_not_write_tombstone_if_value_is_explicitly_unset() {
+    PreparedStatement prepared = session.prepare("INSERT INTO test2 (k, v0) values (?, ?)");
+    BoundStatement boundStatement = prepared.bind(KEY, VALUE);
+    session.execute(boundStatement);
+
+    session.execute(boundStatement.unset("v0"));
+
+    Row row = session.execute("SELECT v0 FROM test2 WHERE k=?", KEY).one();
+    assertThat(row.getInt("v0")).isEqualTo(VALUE);
+  }
+
+  @Test
+  public void should_use_page_size_on_statement() {
+    PreparedStatement prepared =
+        session.prepare(
+            SimpleStatement.newInstance("SELECT v FROM test WHERE k=?").setPageSize(20));
+    BoundStatement statement = prepared.bind(KEY);
+    ResultSet resultSet = session.execute(statement);
+    assertThat(resultSet.getAvailableWithoutFetching()).isEqualTo(20);
+
+    statement = statement.copy(resultSet.getExecutionInfo().getPagingState());
+    resultSet = session.execute(statement);
+
+    assertThat(resultSet.iterator().next().getInt("v")).isEqualTo(20);
+  }
+
+  @Test
+  @Disabled("C* 3.11 returns a ProtocolError, Stargate returns a ServerError")
+  public void should_throw_when_using_corrupt_paging_state() {
+    PreparedStatement prepared = session.prepare("SELECT v FROM test WHERE k=?");
+    BoundStatement statement =
+        prepared
+            .boundStatementBuilder(KEY)
+            .setPagingState(ByteUtils.fromHexString("0x1234"))
+            .build();
+    assertThatThrownBy(() -> session.execute(statement)).isInstanceOf(ProtocolError.class);
+  }
+
+  @Test
+  @Disabled("Looks like the query timestamp is not propagated correctly")
+  public void should_use_query_timestamp() {
```

Comment:
Codacy found an issue: [The JUnit 5 test method name 'should_use_query_timestamp' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/gh/stargate/stargate/pullRequest?prid=6263918)